core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -238,6 +258,9 @@ func (vdb *VersionedDB) getNamespaceDBHandle(namespace string) (*couchDatabase,
 	namespaceDBName := constructNamespaceDBName(vdb.chainName, namespace)
 	vdb.mux.Lock()
 	defer vdb.mux.Unlock()
+	if err := vdb.updateAndSaveNamespaceDBsInfo(namespace, namespaceDBName); err != nil {
+		return nil, err
+	}
```

Comment:
1. Only after we get success from `createCouchDatabase()`, we sould update the channel metadata. Can you place these lines after line 271? 
2. We seem to continue the pattern of side-effect functions. The `createCouchDatabase()` is in fact `createCouchDatabaseIfNotExist()`. For code readers, it is natural to assume that the `createCouchDatabase()` creates a new database. Similarly, when I read `updateAndSaveNamespaceDBsInfo()`, I was wondering why we aren't checking whether the mapping already exists (as createCouchDatabase() does not create a new database always) before performing the update. When I looked at the `updateAndSaveNamespaceDBsInfo()`, I found that we repeat the hidden `if not-exist` pattern. Personally, I would prefer to have the check at the caller before calling the update function (which should blindly update it) similar to `Put()`. If only I find this pattern to add difficulty to code readability, please feel free to ignore this point.

Reply:
For 1, let's discuss the order offline. 
For 2, it is debatable. Personally I prefer the function is self-contained for this kind of check so that it doesn't require/depend on the caller's check.